### ISSUE TITLE
Build fails on both Android and iOS when including SDK

### ISSUE THREAD

**Issue by danielnordh** [RampNetwork/ramp-sdk-flutter#5]
After adding the Ramp SDK our app does not build on either Android or iOS.

**Android**
```
ramp_flutter-2.0.0/android/src/main/kotlin/network/ramp/ramp_flutter/RampFlutterPlugin.kt: (34, 20): 
Object is not abstract and does not implement abstract member public abstract fun onPurchaseFailed(): 
Unit defined in network.ramp.sdk.facade.RampCallback
```
This is after having to modify the AndroidManifest file because Ramp causes a conflict with android:label in `<application>`

**iOS**
```
[!] Unable to find a specification for `Ramp` depended upon by `ramp_flutter`
You have either:
 * out-of-date source repos which you can update with `pod repo update` or with `pod install --repo-update`.
 * mistyped the name or version.
 * not added the source repo that hosts the Podspec to your Podfile.
```
This is after doing a `pod repo update`

**Flutter doctor**
```
Doctor summary (to see all details, run flutter doctor -v):
[✓] Flutter (Channel stable, 3.3.9, on macOS 13.3.1 22E261 darwin-arm, locale en-PT)
[✓] Android toolchain - develop for Android devices (Android SDK version 33.0.0)
[✓] Xcode - develop for iOS and macOS (Xcode 14.3)
[✓] Chrome - develop for the web
[✓] Android Studio (version 2021.3)
[✓] VS Code (version 1.77.3)
```

**Comment by danielnordh** [comment]
cc @jakubsta 
Can you confirm if the package builds for you on either platform?
This is still blocking our off-ramp integration

**Comment by jakubsta** [comment]
Hi @danielnordh ,
A fix for the first issue will be released asap.

Regarding the second issue, have you added `Ramp` to [ios Podfile](https://docs.ramp.network/mobile/flutter-sdk#ios-project)?


Flutter doctor:
```
Doctor summary (to see all details, run flutter doctor -v):
[✓] Flutter (Channel stable, 3.7.12, on macOS 13.3.1 22E261 darwin-arm64, locale en-PL)
[✓] Android toolchain - develop for Android devices (Android SDK version 31.0.0)
[✓] Xcode - develop for iOS and macOS (Xcode 14.3)
[✓] Chrome - develop for the web
[✓] Android Studio (version 2022.1)
[✓] VS Code (version 1.77.3)
[✓] Connected device (3 available)
[✓] HTTP Host Availability

• No issues found!
```

**Comment by danielnordh** [comment]
> Regarding the second issue, have you added `Ramp` to [ios Podfile](https://docs.ramp.network/mobile/flutter-sdk#ios-project)?

I had not, I missed it since it wasn't mentioned on the 'Install' page of pub.dev, and no other Flutter package requires this.

Are there any particular reasons you don't do this the normal Flutter way, including the iOS pod in `flutter_install_all_ios_pods`?
This would simplify both install and future version updates.

**Comment by jakubsta** [comment]
Thanks for the suggestion. We'll check it

**Comment by mateuszjablonski** [comment]
> Are there any particular reasons you don't do this the normal Flutter way, including the iOS pod in `flutter_install_all_ios_pods`?

Hi @danielnordh, can you elaborate on your suggestion? 

Currently, [podspec file](https://github.com/RampNetwork/ramp-sdk-flutter/blob/master/ios/ramp_flutter.podspec) for iOS SDK only contains `s.dependency 'Ramp'`, with no version specified, and it is up to `Podfile` to decide what `Ramp` really means. That's why the line needs to be added to Podfile.
We didn't add a version to podspec nor upload the podspec to Cocoapods trunk, to actively discourage using Cocoapods and promote SPM for native iOS integration. If we added podspec to trunk, and added version to podspec, then iOS pod would be installed without any additional work.

Is there a better way to do it, without adding Ramp to Cocoapods trunk?

**Comment by danielnordh** [comment]
I'm no expert in developing Flutter packages, so don't have any specific implementation proposals.
But I have used a ton of other packages that makes me believe it should be possible to:

- Install ramp_flutter in my pubspec.yaml
- NOT have to add anything in my podfile or use SPM for iOS
- NOT have to edit our AndroidManidest.xml
- Import ONE file, i.e. `import 'package:ramp_flutter/ramp_flutter.dart';` and then be good with any Ramp stuff

It's sort of OK as it is, it just gives the developer more work and more chances things will go wrong.
I would at least update the documentation on pub.dev to include the steps required on iOS, and especially if you recommend using SPM, that was not clear to me so I went ahead and edited the podfile directly like it says in your docs.

Anyway, just trying to help make the package more usable.
